portal/src/model/group.ts

```diff
@@ -1,5 +1,67 @@
+import { LocalValidationError } from "../error/validation";
 import { Group } from "../graphql/adminapi/globalTypes.generated";
 
+export interface CreatableGroup
+  extends Pick<Group, "key" | "name" | "description"> {}
+
+// Ref: pkg/lib/rolesgroups/key.go
+const KEY_REGEX = /^[a-zA-Z_][a-zA-Z0-9:_]*$/;
+const MAX_KEY_LENGTH = 40;
+
+export function generateGroupKeyFromName(name: string): string {
+  let processedName = name.trim();
+  // Replace all whitespace, -, + with _
+  processedName = processedName.replace(/[\s\-+]/g, "_");
+  // Remove all unallowed characters
+  processedName = processedName.replace(/[^a-zA-Z0-9:_]/g, "");
+  // Remove all unallowed prefix characters
+  processedName = processedName.replace(/^[^a-zA-Z_]*/, "");
```

Comment:
Any source for the processing?

Reply:
No source, I just try to convert the name into a format which backend won't return validation error.

Reply:
Ok, will merge first.

@fungc-io See if the processing look ok to you
in some extreme example
name `123 testing +++ $$$$` -> key `_testing_____`

Reply:
ok